pandas/io/excel/_openpyxl.py

```diff
@@ -530,6 +539,20 @@ def _write_cells(
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
 
+        if notes is None or notes_col is None:
+            return
+
+        for row_idx, (_, val) in enumerate(notes.iterrows()):
```

Comment:
iterrows creates a Series per row, which is very expensive. Use `itertuples(index=False)` instead.

Reply:
Thanks for this, I've replaced it and it simplifies the iterating even more now it's just `val` instead of `(_, val)`.